wsdl-core/src/main/java/io/ballerina/wsdl/core/WSDLToBallerina.java

```diff
@@ -0,0 +1,240 @@
+                String typeModulePart = typeGenerator.generate(filteredWSDLOperations);
+                GeneratedSourceFile typesSrc = new GeneratedSourceFile("types.bal", typeModulePart);
+                response.setTypesSource(typesSrc);
+            } catch (FormatterException e) {
+                DiagnosticMessage message = DiagnosticMessage.wsdlToBallerinaErr101(null);
+                diagnosticMessages.add(message);
+            }
+        } catch (WSDLException e) {
+            DiagnosticMessage message = DiagnosticMessage.wsdlToBallerinaErr100(null);
+            diagnosticMessages.add(message);
+        }
+        return DiagnosticUtils.getDiagnosticResponse(diagnosticMessages, response);
+    }
+
+    private void initializeSchemas(Definition wsdlDefinition) {
+        Map<String, XmlSchema> targetNSToSchema = new HashMap<>();
+        Types types = wsdlDefinition.getTypes();
+        List<?> extensions = new ArrayList<>();
+        if (types != null) {
+            extensions = wsdlDefinition.getTypes().getExtensibilityElements();
+        }
+        for (Object extension : extensions) {
+            if (extension instanceof Schema) {
+                Element schElement = ((Schema) extension).getElement();
+                String targetNamespace = schElement.getAttribute("targetNamespace");
+                if (!targetNamespace.isEmpty()) {
+                    targetNSToSchema.put(targetNamespace, new XmlSchemaCollection().read(schElement));
+                }
+            }
+        }
+        SchemaManager.getInstance().initializeSchemas(targetNSToSchema);
+    }
+
+    private WSDLService getSOAPService(Definition wsdlDefinition) {
+        @SuppressWarnings("unchecked")
+        Collection<Service> services = wsdlDefinition.getAllServices().values();
+        for (Service service : services) {
+            @SuppressWarnings("unchecked")
+            Collection<Port> ports = service.getPorts().values();
+            for (Port port : ports) {
+                List<?> extensions = port.getExtensibilityElements();
+                for (Object extension : extensions) {
+                    if (extension instanceof SOAPAddress || extension instanceof SOAP12Address) {
+                        this.soapPort = port;
+                        SOAPVersion version = extension instanceof SOAPAddress ?
+                                SOAPVersion.SOAP11 : SOAPVersion.SOAP12;
+                        String serviceUrl = extension instanceof SOAPAddress ?
+                                ((SOAPAddress) extension).getLocationURI() :
+                                ((SOAP12Address) extension).getLocationURI();
+                        WSDLService wsdlService = new WSDLService();
+                        wsdlService.setSoapVersion(version);
+                        wsdlService.setSoapServiceUrl(serviceUrl);
+                        return wsdlService;
+                    }
+                }
+            }
+        }
+        return null;
```

Comment:
Any reason to return null here ? IMO null should be avoided and use Optionals instead. I think this answer [1] cover the why we should use optionals in a high-level

[1] - https://softwareengineering.stackexchange.com/questions/309134/why-is-using-an-optional-preferential-to-null-checking-the-variable